wig/wig.go

```diff
@@ -119,27 +111,29 @@ func Write(filename string, rec []*Wig) {
 }
 
 //WriteToFileHandle is an helper function for Write that writes the Wig data structure to a file.
-func WriteToFileHandle(file io.Writer, rec *Wig) {
+func WriteToFileHandle(file io.Writer, rec Wig) {
 	var err error
 	if rec.StepType == "fixedStep" {
 		fmt.Println("Printing header")
 		_, err = fmt.Fprintf(file, "%s chrom=%s start=%d step=%d\n", rec.StepType, rec.Chrom,
 			rec.Start, rec.Step)
 		common.ExitIfError(err)
 	} else if rec.StepType == "variableStep" {
-		_, err = fmt.Fprintf(file, "%s chrom=%s", rec.StepType, rec.Chrom)
-		common.ExitIfError(err)
+		log.Fatalf("ERROR: %s is variableStep Wig, must convert to fixedStep before reading in Wig to gonomics", rec.StepType)
 	} else {
 		log.Fatalf("Invalid step type for wig.")
 	}
 
-	for j := range rec.Values {
+	for i := range rec.Values {
 
 		if rec.StepType == "fixedStep" {
-			_, err = fmt.Fprintf(file, "%f\n", rec.Values[j].Value)
-			common.ExitIfError(err)
-		} else if rec.StepType == "variableStep" {
-			_, err = fmt.Fprintf(file, "%d\t%f\n", rec.Values[j].Position, rec.Values[j].Value)
+			if rec.Values[i] == 0 {
```

Comment:
Maybe a more general solution (for all integers instead of only zero) would be:
if rec.Values[i] == float64(int(rec.Values[i])) {
     _, err = fmt.Fprintf(file, "%d\n", int(rec.Values[i]))

Reply:
Would this be converting all values to an integer? 
- If yes, does this effect my ability to do math on these numbers later?  Like if I do math and there is a long decimal, I would want to be able to keep that. But I think all math would be outside of this program (downstream) so maybe it does not matter. 
Or maybe this "typing" is only specific/important to when we are creating the file? And in that case, why would we use [ ]float64 over [ ]integers?

Reply:
This would check to see if the value stored as a float can be stored equally well as an integer.  If so, it would visualize/print the number as an integer, which would not influence any downstream math.  It would just make the file size smaller in many cases by not printing a bunch of trailing zeros.

Reply:
Adding this to the list of convert.go changes.